source/extensions/filters/http/lua/lua_filter.h

```diff
@@ -363,14 +364,24 @@ using FilterConfigConstSharedPtr = std::shared_ptr<FilterConfig>;
 class FilterConfigPerRoute : public Router::RouteSpecificFilterConfig {
 public:
   FilterConfigPerRoute(const envoy::extensions::filters::http::lua::v3::LuaPerRoute& config,
-                       ThreadLocal::SlotAllocator& tls, Api::Api& api);
+                       Server::Configuration::ServerFactoryContext& context);
+
+  ~FilterConfigPerRoute() override {
+    if (per_lua_code_setup_ptr_ && !main_thread_dispatcher_.isThreadSafe()) {
+      main_thread_dispatcher_.post([setup = per_lua_code_setup_ptr_.release()] { delete setup; });
```

Comment:
I agree this is what is needed, but can you add more comments on why this is necessary? Technically I think it's possible this might leak during shutdown but it probably doesn't matter. If you want to avoid a leak no matter what, create a shared_ptr wrapper, put that in the lambda, and then move the unique_ptr inside of it.

Reply:
Thanks. I will add some short comment to explain the necessity and try to fix the possible memory leak.